### DISCUSSION TITLE
Recursively sort object and subobjects

### DISCUSSION THREAD

**Discussion by trust11** [josdejong/svelte-jsoneditor#364] [Ideas]
It would be nice when all elements would be sorted as the function is suggesting to do

**Comment by josdejong** [comment]
Can you explain what you mean with an example? Do you mean recursively sorting all objects for a certain path?

**Comment by trust11** [comment]
Exactly

**Comment by josdejong** [comment]
Right now you can sort the root object via the main menu, and you can sort a nested object/array in `tree` mode by right clicking it, selecting "Sort" from the context menu, and then click the sort button in the popup

There is no API for for recursively sorting multiple objects/arrays in one go. If it is a common request we can consider implementing support for recursively sorting all objects/arrays. What you can do for now is implement this recursive sorting functionality yourself and add a "Sort recursively" button to the main menu via `onRenderMenu`.

**Comment by TonyValenti** [comment]
Yes. I would like a recursive sort as well.
1.  All properties should be sorted by their lowercase name. 
2.  The order of array elements should remain unchanged. 

Thanks so much!

**Comment by josdejong** [comment]
Moving this idea to the [Discussions > Ideas](https://github.com/josdejong/svelte-jsoneditor/discussions/categories/ideas) section since we are not actively working on it. At this point I'm not sure if it is a common enough request to implement it.
